### ISSUE TITLE
[Question and Minor Bug?] any unintended consequences of using separators (e.g. hyphens and underscores) as env var names?

### ISSUE THREAD

**Issue by drmrbrewer** [caprover/caprover#1738]
Over the years, I've ended up with quite a number of environment variables, particularly for one of my apps.  I'm thinking that it might be an idea to add a few "dummy" variables to act as visual separators to organise my env vars into groups.

Something like:

![image](https://user-images.githubusercontent.com/9009792/234665617-4a2165fa-9b4a-4604-94c5-75b74fb0330d.png)

Are there any unintended consequences in CapRover with this approach?  From [here](https://stackoverflow.com/a/2821183/4070848) it seems that any char except `=` is valid, but there is also a suggestion that you should only use letters, digits and underscore.

Incidentally [minor bug?], it seems possible to set duplicate variable names in CapRover via the text field UI:

![image](https://user-images.githubusercontent.com/9009792/234666591-a8199613-f1c1-4c51-a8e0-e076ea1d3253.png)

Switching to Bulk Edit and back again, and these duplicate vars survive, but if you make any edits in Bulk Edit view, the duplicate vars are de-duplicated when switching back to field view.  This seems a little inconsistent?  It should never be possible to enter duplicate var names in either view?

**Comment by githubsaturn** [comment]
> Are there any unintended consequences in CapRover with this approach? 

No, at least not on CapRover. These are simply stored as is and they are passed to Docker as is. Maybe some containers don't like these, but the chance should be next to zero. I don't see why it's could be a problem.


> It should never be possible to enter duplicate var names in either view?

Interesting edge case. What's the suggested solution here? Validate upon type or upon save? As far as I can see, there will be issues with both solutions.

**Comment by drmrbrewer** [comment]
Maybe validation on save, and switching between bulk/field view, and switching between tabs?  But I appreciate that validation gets a bit tricky (e.g. in terms of ideally highlighting where the duplication is).  

But it's a bit disturbing at present when you enter a duplicate var name in bulk edit, then switch back to field view, and only your final one has survived and the rest have just vanished without warning.  

That said, it's probably quite rare that duplicates would occur.  It only happened for me when I was trying out the use of the same name (e.g. `------------`) multiple times for use as a visual separator... I found it was OK via the field view, but not via the bulk view.

Perhaps the simplest would just be to _allow_ duplicate values, i.e. like you can already get via the field-based input view, and make that consistent even via the bulk-edit view?  Then it's just like setting the same environment variable to different values in turn, which isn't illegal in practice, and where only the last one really counts.  Then it's up to the user to get it right.

**Comment by githubsaturn** [comment]
> Perhaps the simplest would just be to allow duplicate values, i.e. like you can already get via the field-based input view, and make that consistent even via the bulk-edit view?

Yea probably this is the best approach
